src/twisted/internet/endpoints.py

```diff
@@ -1466,6 +1467,10 @@ def _parseSSL(
         dhParameters=dhParameters,
         **kw,
     )
+
+    if cipher:
+        if isinstance(cipher, str):
```

Comment:
I guess that it's ok if we don't do type checking for str.


```suggestion
    if cipher is not None:
```

What should happen if you pass a boolean or other types ?

I don't think that we need to worry.

What we need to worry, is to document the `cipher` arument and let people know that it should be `str`

Reply:
[#692e540f](https://github.com/twisted/twisted/pull/12137/commits/692e540f85cac1146fb0a44a230138ca743ce5b0)